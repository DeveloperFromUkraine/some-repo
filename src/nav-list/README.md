## API
    ### Selector
        * mat-nav-list
    
    ### Exported As 
        * matNavList
        
    ### Properties
        * `@Input() disableRipple: boolean`
            * disables/enables rippling effect

## Implementation HTML
    <mat-nav-list>
        <a ign-list-item onClick=window.open() *ngFor="let folder of folders" target="\_blank" tabindex="0">
            <mat-icon mat-list-icon>folder</mat-icon>
            <h4>{{folder.name}}</h4>
            <p> {{folder.updated | date}} </p>
        </a>
    </mat-nav-list>

## Implementation TS
    folders = [
    {
        name: 'Click the folders to open a blank page.',
        updated: new Date('1/1/16'),
    },
    {
        name: 'Your descriptor will go here.',
        updated: new Date('1/17/16'),
    },
    {
        name: 'The dates are just for show.',
        updated: new Date('1/28/16'),
    }
    ];