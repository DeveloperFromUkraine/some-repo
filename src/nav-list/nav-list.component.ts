import { Component } from '@angular/core';

@Component({
    selector: 'ign-nav-list',
    template: `<mat-nav-list><ng-content></ng-content></mat-nav-list>`,
})
export class NavListComponent {}
