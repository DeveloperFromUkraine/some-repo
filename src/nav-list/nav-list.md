*** HTML ***
# Navigation List

## Use
Utilize the mat-nav-list component for navigation lists, such as those with anchor tags. Nav-list components'
    anchor tags can be wrapped in a ign-list-item for multiple links within the list.

## Sample

<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
            <mat-nav-list>
                <a ign-list-item onClick=window.open() *ngFor="let folder of folders" target="\_blank" tabindex="0">
                    <mat-icon mat-list-icon>folder</mat-icon>
                    <h4>{{folder.name}}</h4>
                    <p> {{folder.updated | date}} </p>
                </a>
            </mat-nav-list>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            <p> &lt;mat-nav-list&gt;</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;&lt;a ign-list-item onClick&#61;window&#46;open&#40;&#41; &#42;ngFor&#61;"let folder of folders"</p>
            <p>     target="\_blank" tabindex="0"&gt;</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-icon mat-list-icon&gt;folder&lt;/mat-icon&gt;</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&lt;h4&gt;&#123;folder&#46;name&#125;&lt;/h4&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&lt;p&gt; &#123;folder&#46;updated &#124; date&#125; &lt;/p&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;/a&gt;</p>
            <p> &lt;/mat-nav-list&gt;</p>
        </table>
    </div></mat-tab>
    <mat-tab label="TS"><div class="tab-height">
        <table style="width:100%">
            <p>folders = [</p>
            <p>&nbsp;&nbsp;&#123;</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;name: 'Click the folders to open a blank page.',</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;updated: new Date('1/1/16'), </p>
            <p> &nbsp;&nbsp;&#125;,</p>
            <p> &nbsp;&nbsp;&#123;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;name: 'Your descriptor will go here.',</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;updated: new Date('1/17/16'),</p>
            <p> &nbsp;&nbsp;&#125;,</p>
            <p>&nbsp;&nbsp;&#123;</p>
            <p> &nbsp;&nbsp;&nbsp;&nbsp;name: 'The dates are just for show.',</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;updated: new Date('1/28/16'),</p>
            <p>&nbsp;&nbsp;&#125;</p>
            <p>];</p>
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content

* Mat nav list components differ from list components in that they are used to navigate to other links and pages.
* Mat nav lists should not be used simply declare content in a list form.
* Mat nav lists should contain icons and buttons, as well as descriptors.

## Style

* Mat nav lists should be displayed in vertical positioning.
* Mat nav lists should be spaced evenly apart, to ensure ease of navigation.

## Accessibility

* Add tabindex="0" within the a tag of the mat-nav-list in order to make each link accessible by tab
* Refer to Sample HTML for an example

*** TypeScript *** 
folders = [
        {
            name: 'Click the folders to open a blank page.',
            updated: new Date('1/1/16'),
        },
        {
            name: 'Your descriptor will go here.',
            updated: new Date('1/17/16'),
        },
        {
            name: 'The dates are just for show.',
            updated: new Date('1/28/16'),
        }
    ];

*** ExternalTypeScript ***

*** Imports ***