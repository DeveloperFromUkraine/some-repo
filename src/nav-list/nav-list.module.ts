import { NgModule } from '@angular/core';
import { MatListModule } from '@angular/material';
import { NavListComponent } from './nav-list.component';

@NgModule({
    imports: [
        MatListModule
    ],
    declarations: [
        NavListComponent
    ],
    exports: [
        NavListComponent
    ]
})
export class NavListModule {}
