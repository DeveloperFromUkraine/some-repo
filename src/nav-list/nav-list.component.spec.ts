import { ComponentFixture , TestBed } from '@angular/core/testing';
import { NavListComponent } from './nav-list.component';
import { NavListModule } from './nav-list.module';
import { ComponentTest } from '../../test/test-bed/component';

describe('NavListComponent', () => {
    let fixture: ComponentFixture<NavListComponent>;

    beforeEach(async () => {
        await ComponentTest.createTestBed([NavListModule]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(NavListComponent);
    });

    it('should match snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });
});
