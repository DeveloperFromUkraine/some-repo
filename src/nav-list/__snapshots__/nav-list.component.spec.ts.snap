// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavListComponent should match snapshot 1`] = `
<ign-nav-list>
  <mat-nav-list
    class="mat-nav-list"
    role="navigation"
  />
</ign-nav-list>
`;
